### PR TITLE
Fix: LlamaIndex 0.10.x migration

### PR DESCRIPTION
**Pull request by felipearosr**
Title: Fix: LlamaIndex 0.10.x migration

---

**Description:**

This pull request addresses the issue reported in [Chainlit #752](https://github.com/Chainlit/chainlit/issues/752) regarding migration problems encountered with LlamaIndex version 0.10.x. The main focus of this fix is to ensure compatibility and smooth migration from earlier versions to LlamaIndex 0.10.x without causing disruptions to existing functionalities.

**Changes:**

- Fixed all breaking changes that resulted from deprecated methods and properties in the new version.
- Updated the llama_index test to work with this newer version.

### CODE REVIEW

File: backend/pyproject.toml [hunk 1]
```diff
@@ -53,7 +53,8 @@ optional = true
 [tool.poetry.group.tests.dependencies]
 openai = "^1.11.1"
 langchain = "^0.1.5"
-llama-index = "^0.9.15"
+llama-index = "^0.10.15"
+llama-index-core = "^0.10.15"
```
Comment (tpatel) [inline backend/pyproject.toml:57]: Do we need to install `llama-index-core` for the tests to work? It might be already included in `llama-index`, right?

**Review by tpatel — APPROVED** [review]
Thanks for your contribution ✨

**Comment by vincentclaes** [comment]
Great work! Is there a timeline for the deployment?

**Comment by vincentclaes** [comment]
I checked out the HEAD of main branch, installed that version of chainlit, and I now get

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/vincent/Workspace/rag/venv/lib/python3.11/site-packages/chainlit/__main__.py", line 1, in <module>
    from chainlit.cli import cli
  File "/Users/vincent/Workspace/rag/venv/lib/python3.11/site-packages/chainlit/cli/__init__.py", line 24, in <module>
    from chainlit.server import app, register_wildcard_route_handler
  File "/Users/vincent/Workspace/rag/venv/lib/python3.11/site-packages/chainlit/server.py", line 150, in <module>
    build_dir = get_build_dir("frontend", "frontend")
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vincent/Workspace/rag/venv/lib/python3.11/site-packages/chainlit/server.py", line 147, in get_build_dir
    raise FileNotFoundError(f"{local_target} built UI dir not found")
FileNotFoundError: frontend built UI dir not found
```

**Comment by vincentclaes** [comment]
I guess this has to do that the "frontend" folder is not taken into account when I create a wheel of the backend folder and install that. I think I will wait for the official release :-)

**Comment by tpatel** [comment]
You can follow the "contributing" document to [launch Chainlit from source](https://github.com/Chainlit/chainlit/blob/main/.github/CONTRIBUTING.md#start-the-chainlit-server-from-source). I believe you're missing the frontend compilation step `pnpm run buildUi` to get it to work.

**Pull request merged**